components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverUsernameApiServiceImpl.java

```diff
@@ -48,6 +48,13 @@ public Response recoverUsernamePost(List<UserClaimDTO> claim, String tenantDomai
         } catch (IdentityRecoveryException e) {
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
+            if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    throwable.getMessage())) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(throwable.getMessage());
```

Comment:
Why are we not loggin the thowable... We need to log full stacktrace.

Reply:
Also, how do we konw this is happening due to **email not found only**?

Reply:
We are logging throwable separately. This is for "email not found" case. I will add the full stack trace to the log.